### DISCUSSION TITLE
What happens to the Kafka table?

### DISCUSSION THREAD

**Discussion by Zonalds** [ClickHouse/ClickHouse#49123] [Q&A] [answered]
In the documentation, one can integrate Kafka with Clickhouse this way;

```
  CREATE TABLE queue (
    timestamp UInt64,
    level String,
    message String
  ) ENGINE = Kafka('localhost:9092', 'topic', 'group1', 'JSONEachRow');

  CREATE TABLE daily (
    day Date,
    level String,
    total UInt64
  ) ENGINE = SummingMergeTree(day, (day, level), 8192);

  CREATE MATERIALIZED VIEW consumer TO daily
    AS SELECT toDate(toDateTime(timestamp)) AS day, level, count() as total
    FROM queue GROUP BY day, level;

  SELECT level, sum(total) FROM daily GROUP BY level;
```

What happens to the Kafka (queue) table afterwards? Does it store data too or delete them after they have been ingested on the daily table?

**Comment by tsolodov** [comment, marked as answer]
The `queue` table is not meant to store data, but rather to serve as a Kafka consumer that reads data from the topic on localhost:9092, using the consumer group `group1`. Each time data is read from the queue table, the offset of the consumer is incremented.
